### PR TITLE
mqtt: Improve handling of multiple PDU parsing

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10236

Issue: 6592

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6592](https://redmine.openinfosecfoundation.org/issues/6592)

Describe changes:
- Parse PDU instead of entire stream

Updates:
- Reflect remaining byte count (per review comment)
- s-v test pr update

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1609
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17783

**Comment by glongo** [comment]
According to Victor's suggestion (https://github.com/OISF/suricata/pull/10058#discussion_r1430988421):

> the frame should be created earlier, and not depend on the parser returning ok. It should be created as soon as we start parsing the data, and then be updated with the length when it's complete

For reference, you can look here for reference: 

https://github.com/OISF/suricata/pull/10098/commits/73665068e233452a4f669118182bc8f5535958be#diff-d988869639f4f6ed35c7047322b592a33a7421d22c38aa6137d456967b3656d3R162-R200

File: rust/src/mqtt/mqtt.rs [hunk 1]
```diff
@@ -462,13 +462,13 @@ impl MQTTState {
...
-            SCLogDebug!("request: handling {}", current.len());
+            SCLogDebug!("request: handling {}; rem: {}", current.len(), input.len() - current.len());
             match parse_message(current, self.protocol_version, self.max_msg_len) {
                 Ok((rem, msg)) => {
                     let _pdu = Frame::new(
                         flow,
                         &stream_slice,
-                        input,
+                        current,
                         current.len() as i64,
```
Comment (catenacyber) [inline rust/src/mqtt/mqtt.rs:472]: Should it not be `current.len()-rem.len()` ? (and same below)

**Comment by jlucovsky** [comment]
Continued in #10279

File: rust/src/mqtt/mqtt.rs (hunk 1 above)
Comment (jlucovsky) [reply rust/src/mqtt/mqtt.rs:472]: Will adjust in new pr

**Pull request closed without merging**